src/herder/TransactionQueue.cpp

```diff
@@ -1167,24 +1184,20 @@ SorobanTransactionQueue::broadcastSome()
 size_t
 SorobanTransactionQueue::getMaxQueueSizeOps() const
 {
-    if (protocolVersionStartsFrom(mApp.getLedgerManager()
-                                      .getLastClosedLedgerHeader()
-                                      .header.ledgerVersion,
-                                  SOROBAN_PROTOCOL_VERSION))
-    {
-        auto res = mTxQueueLimiter->maxScaledLedgerResources(true);
-        releaseAssert(res.size() == NUM_SOROBAN_TX_RESOURCES);
-        return res.getVal(Resource::Type::OPERATIONS);
-    }
-    else
-    {
-        return 0;
-    }
+    std::lock_guard<std::recursive_mutex> guard(mTxQueueMutex);
+    // TODO: I removed a conditional checking that the protocol version is
```

Comment:
I _think_ this could cause a crash if we start a node from scratch and try to catchup, since it'll be in protocol 0. Any reason to remove it at this time?

Reply:
Ah, good point. I re-added the check. No particular reason to remove, other than that I was auditing uses of `mApp` and failed to think about the "catchup from 0" case here.